src/eradiate/test_tools.py

```diff
@@ -71,3 +75,142 @@ def missing_artefact(filename: PathLike) -> None:
+
+    return rmse <= rmse_threshold, rmse
+
+
+def assert_chi2(reference, result, significance=0.05):
+    from mitsuba.core.math import chi2
+    from mitsuba.python.math import rlgamma
+
+    histo_bins = np.linspace(reference.min(), reference.max(), 20)
+    histo_ref = np.histogram(reference, histo_bins)[0]
+    histo_res = np.histogram(result, histo_bins)[0]
+
+    # sorting both histograms following the ascending frequencies in
+    # the reference. Algorithm from:
+    # https://stackoverflow.com/questions/9764298/how-to-sort-two-lists-which-reference-each-other-in-the-exact-same-way
+    histo_ref_sorted, histo_res_sorted = zip(
+        *sorted(zip(histo_ref, histo_res), key=lambda x: x[0])
+    )
+
+    chi2val, dof, pooled_in, pooled_out = chi2(histo_res_sorted, histo_ref_sorted, 5)
+
+    p_value = 1 - rlgamma(dof / 2.0, chi2val / 2.0)
+
+    if p_value < significance:
+        passed = False
+    else:
+        passed = True
+
+    return passed, p_value
+
+
+def create_archive_files(path: PathLike, filename: str, data: xr.Dataset):
+    """
+    Create NetCDF output files for regression tests to be archived.
+
+    Parameters
+    ----------
+
+    path : path-like
+        Path of the directory to store the NetCDF file.
+        If the directory does not exist, attempt to create it
+
+    filename: str
+        Filename for result storage
+
+    data: xarray.Dataset
+        Result data to be stored in a NetCDF file
+    """
+
+    if not os.path.exists(path):
+        os.makedirs(path)
+
+    data.to_netcdf(os.path.join(path, filename))
+
+
+def create_regression_test_plots(data: xr.Dataset, filename: PathLike, metric: tuple):
+    """
+    ...
+
+    Parameters
+    ----------
+
+    data : xarray.Dataset
+        Dataset containing the reference and test result
+
+    filename : path-like
+        Absolute path to the output file for the plot
+
+    metric : tuple
+        A tuple of the form (metric name, value) to be added to the plots.
+    """
+    eradiate.plot.set_style()
+    fig, axes = plt.subplots(2, 2, figsize=(8, 6), dpi=120)
+
+    ref = np.squeeze(data.reference)
+    result = np.squeeze(data.brf)
+    vza = data.vza
+    axes[0][0].plot(vza, ref, label="reference")
+    axes[0][0].plot(vza, result, label="result")
+    axes[0][0].set_title("Reference and test result")
+    handles, labels = axes[0][0].get_legend_handles_labels()
+
+    axes[1][0].plot(vza, result - ref)
+    axes[1][0].set_xlabel("VZA in degrees")
+    axes[1][0].set_ylabel("BRFpp")
+    axes[1][0].set_title("Absolute difference")
+
+    axes[1][1].plot(vza, (result - ref) / ref)
+    axes[1][1].set_title("Relative difference")
+
+    axes[0][1].set_axis_off()
+    axes[0][1].legend(handles=handles, labels=labels, loc="upper center")
+    axes[0][1].text(
+        0.5, 0.5, f"{metric[0]}: {metric[1]:.4}", horizontalalignment="center"
+    )
+
+    plt.tight_layout()
+    plt.savefig(filename)
```

Comment:
Don't forget to close that figure:
```suggestion
    plt.savefig(filename)
    plt.close(fig)
```